console/frontend/src/main/frontend/src/app/views/storage/storage-list/storage-list.component.ts

```diff
@@ -127,81 +127,83 @@ export class StorageListComponent implements OnInit, AfterViewInit {
     );
     // this.$state.current.data.breadcrumbs = "Adapter > " + (this.$state.params["storageSource"] == 'pipes' ? "Pipes > " + this.$state.params["storageSourceName"] + " > " : "") + this.$state.params["processState"] + " List";
 
-    this.dtOptions = {
-      searching: false,
-      scrollX: true,
-      // bAutoWidth: false,
-      autoWidth: false,
-      orderCellsTop: true,
+    this.datasource.options = {
+      filter: false,
       serverSide: true,
-      processing: true,
-      paging: true,
-      lengthMenu: [10, 25, 50, 100, 500, 999],
-      order: [[3, 'asc']],
-      columns: this.initialColumns,
-      // sAjaxDataProp: 'messages',
-      ajax: (data, callback): void => {
-        const dataRecord = data as Record<string, NonNullable<unknown>>;
-        const start = dataRecord['start'];
-        const length = dataRecord['length'];
-        const order = (dataRecord['order'] as unknown[])[0] as { dir: 'asc' } | { dir: 'desc' };
-        const direction = order.dir;
-
-        let queryParameters = `?max=${length}&skip=${start}&sort=${direction}`;
-        const search = this.search;
-        const searchSession: Record<keyof typeof search, string> = {};
-        for (const column in search) {
-          const text = search[column as keyof typeof search];
-          if (text) {
-            queryParameters += `&${column}=${text}`;
-            searchSession[column as keyof typeof search] = text;
+    };
+    this.datasource.setServerRequest(
+      (requestInfo) =>
+        new Promise((resolve, reject) => {
```

Comment:
I think this would be more readable if it were a couple of seperate methods. 
Even for the subscribe.
Something like `subscribe({next: resolveMessages, error: handleError});`

Reply:
Solved as discussed together irl